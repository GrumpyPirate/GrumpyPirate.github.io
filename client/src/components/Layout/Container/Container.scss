@import 'core';

.container {
  width: 100%;
  padding-right: rem(map-get($gutter-widths, xs) / 2);
  padding-left: rem(map-get($gutter-widths, xs) / 2);

  @each $breakpoint in map-keys($gutter-widths) {
    @if index($breakpoints, $breakpoint map-get($breakpoints, $breakpoint)) > 1 {
      @include respond-up($breakpoint) {
        padding-right: rem(map-get($gutter-widths, $breakpoint) / 2);
        padding-left: rem(map-get($gutter-widths, $breakpoint) / 2);
      }
    }
  }

  // Modifiers
  &--is {
    &-fluid {
      width: 100%;
    }
  }
}
