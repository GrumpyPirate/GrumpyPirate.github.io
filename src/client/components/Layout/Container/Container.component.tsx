import React, { FunctionComponent } from 'react';

import { ChildrenProps } from 'types';

import classes from './Container.component.scss';

const Container: FunctionComponent<ChildrenProps> = ({ children }: ChildrenProps) => (
  <div className={classes['container']}>{children}</div>
);

export default Container;
