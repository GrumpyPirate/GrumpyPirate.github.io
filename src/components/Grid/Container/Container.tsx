import styled from 'styled-components';

import { gutterWidths, mediaQueries, rem } from 'styles';

const Container = styled.div`
  width: 100%;
  padding-right: ${rem(gutterWidths.xs / 2)};
  padding-left: ${rem(gutterWidths.xs / 2)};

  @media ${mediaQueries.md} {
    padding-right: ${rem(gutterWidths.md / 2)};
    padding-left: ${rem(gutterWidths.md / 2)};
  }
`;

export default Container;
