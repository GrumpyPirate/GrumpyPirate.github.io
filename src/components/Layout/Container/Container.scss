@import 'core';

.container {
  padding-left: rem(map-get($gutter-widths, 'xs') / 2);
  padding-right: rem(map-get($gutter-widths, 'xs') / 2);
  width: 100%;

  @each $breakpoint in map-keys($gutter-widths) {
    @if index($breakpoints, $breakpoint map-get($breakpoints, $breakpoint)) > 1 {
      @include respond-up($breakpoint) {
        padding-left: rem(map-get($gutter-widths, $breakpoint) / 2);
        padding-right: rem(map-get($gutter-widths, $breakpoint) / 2);
      }
    }
  }

  // Modifiers
  &--is {
    &-fluid {
      width: 100%;
    }
  }
}
