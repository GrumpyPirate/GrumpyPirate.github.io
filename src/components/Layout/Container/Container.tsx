import React, { FunctionComponent } from 'react';

import { ChildrenProps } from 'types/common';

import classes from './Container.scss';

const Container: FunctionComponent<ChildrenProps> = ({ children }) => (
  <div className={classes['container']}>{children}</div>
);

export default Container;
