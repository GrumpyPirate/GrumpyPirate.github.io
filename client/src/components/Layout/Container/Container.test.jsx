import React from 'react';
import Container from './Container';

describe('<Container />', () => {
  const props = {
    children: <div>Test children</div>,
  };

  it('should exist', () => {
    expect(Container).toBeDefined();
  });

  it('should be a function', () => {
    expect(typeof Container).toBe('function');
  });

  it('should render as expected, without crashing', () => {
    expect(<Container {...props} />).toMatchSnapshot();
  });
});
