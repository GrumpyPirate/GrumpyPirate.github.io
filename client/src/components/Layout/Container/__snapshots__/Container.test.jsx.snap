// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Container /> should render as expected, without crashing 1`] = `
<Container>
  <div>
    Test children
  </div>
</Container>
`;
